### PR TITLE
fix(Megamenu): add margin to prevent shifting on scroll lock

### PR DESCRIPTION
**Pull request by annawen1**
### Related Ticket(s)

Masthead Navigation moving on toggle #5472

### Description

For users on windows machines or have set the scrollbar to always show on their Macs, the content and masthead will shift when opening and closing the megamenu. This is due to the scrollbar lock when the megamenu is open, removing the scrollbar. The content then shifts due to the scrollbar disappearing.

Solution is to get the width of the scrollbar and then add the width to the body and masthead as `margin-right`.

BEFORE:
before

AFTER:
after

### Changelog

**Changed**

- make sure the `body` and masthead add a margin-right of the scrollbar width when the megamenu is expanded to prevent content from shifting

### CODE REVIEW

**Review by IgnacioBecerra — APPROVED** [review]
LGTM, thanks @annawen1 !!

**Comment by ariellalgilmore** [comment]
React looks good to me!

I just tested web components though and still seeing the shifting

https://user-images.githubusercontent.com/20210594/117210191-36c82a80-adac-11eb-870e-d2dddac9e2bf.mov

**Comment by annawen1** [comment]
> React looks good to me!
> 
> I just tested web components though and still seeing the shifting
> 
>  Screen.Recording.2021-05-05.at.2.12.34.PM.mov

For some reason, the deploy preview for web-components keeps failing with the newest changes I put in. Looking into it

File: packages/web-components/src/components/masthead/megamenu-top-nav-menu.ts [hunk 1]
```diff
@@ -91,8 +97,24 @@ class DDSMegaMenuTopNavMenu extends DDSTopNavMenu {
...
+        doc.body.style.marginRight = `${this._scrollBarWidth}px`;
+        forEach(doc.querySelectorAll((this.constructor as typeof DDSMegaMenuTopNavMenu).selectorOverlay), item => {
+          (item as DDSMegaMenuOverlay).active = this.expanded;
+        });
+        // add the scrollbar width as right-margin to prevent content from shifting when
+        // scrollbar disappears on megamenu expand
+        const masthead = <HTMLElement>doc.querySelector('dds-masthead')?.shadowRoot?.querySelector('.bx--masthead__l0');
+        if (masthead) {
+          masthead.style.marginRight = `${this._scrollBarWidth}px`;
+        }
```
Comment (kennylam) [inline packages/web-components/src/components/masthead/megamenu-top-nav-menu.ts:109]: @annawen1 Looks like this extra `}` is the cause of the Jenkins build error.

Comment (kennylam) [reply packages/web-components/src/components/masthead/megamenu-top-nav-menu.ts:109]: Never mind, disregard. I didn't see that it's nested above.

**Review by ariellalgilmore — APPROVED** [review]
LGTM

**Pull request merged**